dojo/tools/cobalt_api/parser.py

```diff
@@ -0,0 +1,182 @@
+import json
+from datetime import datetime
+from dojo.models import Endpoint, Finding
+from dojo.tools.cobalt_api.importer import CobaltApiImporter
+
+__author__ = 'ericcornelissen'
+
+
+class CobaltApiParser(object):
+    """
+    Import from Cobalt.io API /findings
+    """
+
+    def get_scan_types(self):
+        return ["Cobalt.io API Import"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "Cobalt.io API Import"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Cobalt.io API findings can be imported in JSON format (option --json)."
+
+    def get_findings(self, file, test):
+        if file is None:
+            data = CobaltApiImporter().get_findings(test)
+        else:
+            data = json.load(file)
+
+        findings = []
+        for entry in data["data"]:
+            resource = entry["resource"]
+            links = entry["links"]
+            if not self.include_finding(resource):
+                continue
+
+            cobalt_state = resource["state"]
+            cobalt_severity = resource["severity"]
+            cobalt_log = resource["log"]
+
+            title = resource["title"]
+            date = self.get_created_date(cobalt_log)
+            description = "\n".join(
+                [
+                    resource["description"],
+                    "",
+                    "Cobalt.io details:",
+                    f"- Impact: {resource['impact']}",
+                    f"- Likelihood: {resource['likelihood']}",
+                    "",
+                    "Cobalt.io link:",
+                    links["ui"]["url"],
+                ]
+            )
+            mitigation = resource["suggested_fix"]
+            steps_to_reproduce = resource["proof_of_concept"]
+            endpoints = resource["affected_targets"]
+            last_status_update = self.get_latest_update_date(cobalt_log)
+            unique_id_from_tool = resource["id"]
+
+            finding = Finding(
+                test=test,
+                title=self.safe_title(title),
+                date=date,
+                severity=self.convert_severity(cobalt_severity),
+                description=description,
+                mitigation=mitigation,
+                steps_to_reproduce=steps_to_reproduce,
+                active=self.is_active(cobalt_state),
+                verified=self.is_verified(cobalt_state),
+                false_p=self.is_false_p(cobalt_state),
+                duplicate=self.is_duplicate(cobalt_state),
+                out_of_scope=self.is_out_of_scope(cobalt_state),
+                risk_accepted=self.is_risk_accepted(cobalt_state),
+                is_mitigated=self.is_mitigated(cobalt_state),
+                last_status_update=last_status_update,
+                numerical_severity=self.convert_severity_numerical(cobalt_severity),
+                static_finding=False,
+                dynamic_finding=False,
```

Comment:
How is it possible? Most of parsers are static or dynamic. Does Cobalt mix the both?

Reply:
I was thinking that Cobalt.io is a "Pentest as a service" so dynamic?

Reply:
That makes sense to me, I updated it.

To clarify, I initially opted to set `dynamic_finding` to `False` as well since I wasn't sure a pentest counts as a "DAST **tool**".

https://github.com/DefectDojo/django-DefectDojo/blob/3f443735126eb987cb7d8e47fb54f427cd8652bd/dojo/models.py#L1821-L1823

Reply:
I guess, maybe a third flag is missing here, like `pentest_finding`.

Reply:
That sounds like a better solution, should I add that as part of this Pull Request?

Reply:
no, it will be too big. let's use `dynamic_finding` and we will work on another flag another time.

Reply:
Had some discussion with the team, use `dynamic_finding`.